### PR TITLE
[WIP] Very opinionated changes to the framework

### PR DESCRIPTION
**Pull request by Mistobaan**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -16,21 +20,37 @@ To submit a bug report or a feature request, please open an [issue](https://gith
 
 Follow these steps to start contributing code:
 
+1. Setup your environment i.e:
```
Comment (jon-tow) [inline CONTRIBUTING.md:23]: Maybe we should add this to the dependencies steps? https://github.com/CarperAI/trlx/blob/974d9a975c51805dc8d3ac141090515692657642/CONTRIBUTING.md?plain=1#L29-L33

What do you think?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -16,21 +20,37 @@ To submit a bug report or a feature request, please open an [issue](https://gith
...
+conda create -n trlx python=3.8 -c pytorch
+```
 1. Create your own [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo#forking-a-repository) of the repository and clone it to your local machine.
     ```bash
     git clone https://github.com/<YOUR-USERNAME>/trlx.git
     cd trlx
     git remote add upstream https://github.com/CarperAI/trlx.git
     ```
-2. Create a new branch for your changes and give it a concise name that reflects your contribution.
+1. Create a new branch for your changes and give it a concise name that reflects your contribution.
```
Comment (jon-tow) [inline CONTRIBUTING.md:33]: Seems like the numbering got messed up here

File: setup.py [hunk 3]
```diff
@@ -1,3 +1,87 @@
...
+        "pyyaml",
+        "torch~=1.13.0",
+        "accelerate~=0.15.0",
+        "typing-extensions~=3.10.0",
+        "torchtyping~=0.1.4",
+        "ray"
+    ],
+    extras_require=extras,
+    classifiers=[
+        "Development Status :: 4 - Production/Stable",
```
Comment (jon-tow) [inline setup.py:57]: We're still in alpha so let's change this classifier to

```
"Development Status :: 3 - Alpha",
```

File: setup.py [hunk 4]
```diff
@@ -1,3 +1,87 @@
-from setuptools import setup
+from setuptools import find_packages, setup
```
Comment (jon-tow) [inline setup.py:1]: Curious, what's the benefit of switching to `setup.py` for handling the project metadata instead of the previous `setup.cfg`/`pyproject.toml`?

File: setup.py [hunk 5]
```diff
@@ -1,3 +1,87 @@
...
+    install_requires=[
+        "numpy>=1.17",
+        "packaging>=20.0",
+        "psutil",
+        "pyyaml",
+        "torch~=1.13.0",
+        "accelerate~=0.15.0",
+        "typing-extensions~=3.10.0",
+        "torchtyping~=0.1.4",
+        "ray"
```
Comment (jon-tow) [inline setup.py:53]: Do we need to duplicate the requirements that already exist in the `setup.cfg` metadata?

File: examples/ppo_sentiments.py [hunk 6]
```diff
@@ -18,7 +18,7 @@ def get_positive_score(scores):
     return dict(map(lambda x: tuple(x.values()), scores))["POSITIVE"]
 
 
-default_config = yaml.safe_load(open("configs/ppo_config.yml"))
+default_config = yaml.safe_load(open("../configs/ppo_config.yml"))
```
Comment (jon-tow) [inline examples/ppo_sentiments.py:21]: Revert this. We've been standardized on accessing paths from the root so we'd need to change this across all other examples. We can do this in a separate PR if you think it's worth it!

**Review by jon-tow — CHANGES_REQUESTED** [review]
Thanks for the changes @Mistobaan ! I know this is a draft but I've left some comments if you could consider them :)

File: CONTRIBUTING.md (hunk 2 above)
Comment (Mistobaan) [reply CONTRIBUTING.md:33]: Markdown handles it automatically so you don't have to change it. 
```
1. is one
1. is one
1. Is one
```
1. is one
1. is one
1. Is one
